server/server.go

```diff
@@ -594,6 +599,15 @@ func (s *Server) AcceptLoop(clr chan struct{}) {
 	for s.isRunning() {
 		conn, err := l.Accept()
 		if err != nil {
+			if s.isLameDuckMode() {
```

Comment:
Why would we not shut down the listeners by closing the socket vs this approach?

Reply:
ok I see you use this to catch the state when you close the socket.

Reply:
When closing the listener, we would get an error. If we don't check for that and since server isRunning() is still true, we would go back in an accept, and fail again, etc.. More over, you don't want AcceptLoop to return right away because otherwise Start() will exit which means that the process will quit before we had a chance to do anything.

Reply:
Gotcha..